src/node/coinstats.cpp

```diff
@@ -60,11 +73,23 @@ static bool GetUTXOStats(CCoinsView* view, CCoinsStats& stats, T hash_obj, const
     stats = CCoinsStats();
     std::unique_ptr<CCoinsViewCursor> pcursor(view->Cursor());
     assert(pcursor);
-
     stats.hashBlock = pcursor->GetBestBlock();
+
+    const CBlockIndex* block_index;
     {
         LOCK(cs_main);
-        stats.nHeight = LookupBlockIndex(stats.hashBlock)->nHeight;
+        block_index = LookupBlockIndex(pcursor->GetBestBlock());
```

Comment:
5479000c79b254120b7535bf322869439ee33215: consider splitting the usage of the index away from enabling the index

Reply:
I added an option (`no_index`) for it in a new commit.